[depends] set CMAKE_SYSTEM_PROCESSOR on Toolchain_*.cmake.in

AlwinEsch opened the pull request:
## Description

Some addon dependencies built with cmake need the CMAKE_SYSTEM_PROCESSOR to prepare for the desired OS.

This has not been handed over so far, only the self-defined variable "OS" was available, which is an unknown definition name for the cmake based dependencies.

The [CMAKE_SYSTEM_PROCESSOR](https://cmake.org/cmake/help/v3.7/variable/CMAKE_SYSTEM_PROCESSOR.html) variable is hereby also transferred.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
Related to this request https://github.com/xbmc/imagedecoder.mpo/pull/8
Tested on Mac with OSX and IOS build and on Linux x86_64.

<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Rechi reviewed (changes requested):
No, extending the args passed for configuring addon dependencies is wrong.
Set it in the toolchain configration file.

AlwinEsch commented:
Is updated and becomes set on `Makefile.include.in` to the wanted CMAKE
```make
CMAKE=@prefix@/@tool_dir@/bin/cmake -DCMAKE_TOOLCHAIN_FILE=$(PREFIX)/share/Toolchain.cmake -DCMAKE_INSTALL_PREFIX=$(PREFIX) -DCMAKE_SYSTEM_PROCESSOR=$(CPU)
```

However, by external project the CMAKE_SYSTEM_PROCESSOR must be explicitly passed on, otherwise it would only be available in the base.

Rechi reviewed (changes requested):
No, extending the args passed for configuring addon dependencies is wrong.
Set it in the toolchain configration file.

AlwinEsch commented:
I thought on begin you mean with **"toolchain configration file"** the `/tools/depends/Makefile.include.in` where also is with `*.in` a configuration related file for me.
And not looked on `target` folder about `Toolchain_binaddons.cmake.in` (had overlooked it because it was at the bottom of the folder list and has not scrolled down :smirk:) . Thing you mean this, should then also the `Toolchain.cmake.in` changed?

AlwinEsch commented:
Is updated and test worked. Just overlooked these files. 

Now only for addons, since Kodi works and the request primarily relates to addons.

Rechi commented:
Why only for addons?
At which place, where `Toolchain.cmake` is used, do you think setting the variable is incorrect?

AlwinEsch commented:
I add it there too.

Added them as question above:
```
Thing you mean this, should then also the Toolchain.cmake.in changed?
```

AlwinEsch commented:
Is updated, about **`At which place, where Toolchain.cmake is used, do you think setting the variable is incorrect?`** seen it not as incorrect (is incorrect without them defiend), only was a bit angry about a backport to Leia of them to change if it already works (not sure why, maybe unused).

Rechi commented:
Have you checked the value of `CMAKE_SYSTEM_PROCESSOR` for the different systems and are they appropriate?

AlwinEsch commented:
I checked the available OS (Linux and Mac) with normal x86_64, iOS and Raspberry build.
Few things use the cmake and this value, not even the addon because it wasn't available there since the beginning.
But find it very appropriate to always have this value available because it is worth a standard CMake basis, which should always be there.

Rechi commented:
Yes, not always setting the variable would be wrong.
But for one of the platforms you checked, the value of the variable (CMAKE_SYSTEM_**PROCESSOR**), doesn't look right.

AlwinEsch commented:
Yes, you seem to be right. The `cortex-a7` would not really be correct for CMake on the Raspberry, should be only `arm`.

It's a bit stupid to find a complete list of standard values on the web. I'll keep looking, let's see if there is anything in this regard to get CMake's own. And these translate that appropriately, there must be something there since we are not the only ones with such problems.

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -266,6 +270,13 @@ case $host in
+        if test "x$use_cpu" = "xauto"; then
+          use_cpu=x86_64
+        fi
+        use_cmake_cpu=x86_64
```
This `x86_64*-linux-android*)` was missing before

Rechi replied:
How is adding a new platform which currently isn't support, related to this PR?

AlwinEsch replied:
Becomes separated, but makes no sense to define a i686 for Android only and no x86_64 where is mostly the standard of them (by x86 CPU way)

And https://github.com/xbmc/xbmc/blob/master/docs/README.Android.md not correct, this makes on my system a 64bit and if I take a very old PC a 32bit

Rechi replied:
Where does the the android documentation mention a x86_64 host?
How is your x86_64 build platform related to the different build platforms?

AlwinEsch replied:
One is here https://gitlab.kitware.com/cmake/cmake/blob/v3.13.0/Modules/Platform/Android-Determine.cmake#L203, the `_ANDROID_ABI_x86_64_PROC` and `_ANDROID_ABI_x86_PROC`
And the https://developer.android.com/ndk/guides/standalone_toolchain 

Also how we use `host_cpu` and `./configure --host=...` seems not really correct. 
Host is for me my base where `x86_64` and normally by `configure` it should use `--target=...` (or I think bad :thinking:)

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -281,6 +292,11 @@ case $host in
+    if [[ $host =~ "arm64-" ]] || [[ $host =~ "aarch64-" ]]; then
+      use_cmake_cpu=aarch64
+    else
+      use_cmake_cpu=arm
```
For linux I set it to `arm` alone, not sure how there could be easiest (without big checks) the right defined.

Rechi replied:
There is a variable which already contains exactly one of the two values, depending on the host used to configure. So why not use that one?

AlwinEsch replied:
You mean the `arm*` from `arm*-*-linux-gnu*`, can look to extract them from `host`, does it then match the mostly on cmake used variables?

AlwinEsch replied:
The another thing it can also `aarch32` which then not correct to cmake.

AlwinEsch replied:
I add this:
```bash
    if [[ $host =~ "arm64-" ]] || [[ $host =~ "aarch64-" ]]; then
      use_cmake_cpu=aarch64
    elif [[ $host =~ "arm32-" ]] || [[ $host =~ "aarch32-" ]]; then
      # aarch32 related to https://en.wikipedia.org/wiki/ARM_architecture#AArch32 is v7-a
      # arm32 should never be there, only in the check to avoid possible wrong cmake CPU
      use_cmake_cpu=armv7-a 
    else
      use_cmake_cpu=$(echo $host | sed -e 's/\-.*$//')
    fi
```

Then the cpu becomes taken from host, if not `arch*` or `32|64`

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -319,6 +340,9 @@ case $host in
           use_cpu=x86_64
+          use_cmake_cpu=x86_64
+        else
+          use_cmake_cpu=$use_cpu
```
Not idea which can be a alternative CPU to x86_64-apple-darwin, it set here the `use_cmake_cpu` to this from `use_cpu`.

Maybe hardcode there `use_cmake_cpu=x86_64` for all?

Rechi replied:
How would setting the custom passed cpu value instead of x86_64 be correct for `CMAKE_SYSTEM_PROCESSOR` for this host?

AlwinEsch replied:
Therefore this place also confused me, it doesn't really make sense to have another CPU there (about `if test "x$use_cpu" = "xauto"; then`).

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -347,12 +371,16 @@ case $host in
             use_cpu=armv7
+            use_cmake_cpu=armv7
+          else
+            use_cmake_cpu=$use_cpu
```
Becomes for iOS 64bit always the `--with-cpu` given by configure? 
Was confused that there no `armv8` or something else becomes defined.

AlwinEsch replied:
Seen the `--with-cpu=arm64` becomes used, I add a check for this name and set it then to `aarch64` for `use_cmake_cpu`, any different `--with-cpu=` there possible?

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -437,6 +467,7 @@ case $use_platform in
   raspberry-pi3)
      target_platform=raspberry-pi
      use_cpu=cortex-a53
+     use_cmake_cpu=aarch64
```
Here and above becomes related to Raspberry the correct cmake cpu set

Rechi replied:
Are you sure this is correct?

AlwinEsch replied:
About them https://en.wikipedia.org/wiki/ARM_Cortex-A53 is it only 64bit. Is it possible to use this a 32bit?

Rechi replied:
Yes the cpu might be 64 Bit. Should this value really reflect the architecture of the Cortex-A53 processor, or wouldn't be the targeting architecture of the produced binaries the only correct value?

AlwinEsch replied:
Have looked a little more so more would be wrong. The `Cortex-A53` is used from Version Mod. B v1.2 (Cortex-A7, 32bit), the older have (ARM1176JZF-S, 32bit) and the new have Cortex-A53 64bit. A bit hard mix in one Version

AlwinEsch replied:
They use mostly only `armv7-a` the 64bit is available, but mostly still not supported on Raspberry.

AlwinEsch reviewed (commented):
Have updated, not complete tested now. Want to make sure that this is a correct way.

The `use_cmake_cpu` becomes then internally by the `configure` set and then used on Toolchain.

EDIT: Also another idea for a better name as `use_cmake_cpu`?

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -266,6 +270,13 @@ case $host in
         if test "x$use_cpu" = "xauto"; then
           use_cpu=$host_cpu
```
Is this `$host_cpu` really correct, make no sense on i*86 (32bit) to use my system 64bit :sweat_smile:

Rechi replied:
What's the value of `host_cpu`?

Rechi commented on tools/depends/configure.ac:
```diff
@@ -248,6 +248,9 @@ case $host in
           use_cpu="armeabi-v7a"
+          use_cmake_cpu=armv7-a
+        else
+          use_cmake_cpu=armv6
```
Is armv6 still supported?

AlwinEsch replied:
Was thought as minimum callback, heard armv6 is almost identical to armv7.

Rechi replied:
But does Kodi (and all it's dependencies) still support armv6?

AlwinEsch replied:
I can leave away, no problem, what you thing is better.

To force `use_cmake_cpu=armv7-a` for all (where I think then the best) or make `use_cmake_cpu=$host_cpu`?

Is confusing for me that there still `use_cpu` is checked for `auto` then.

AlwinEsch replied:
Becomes forced to `armv7-a`, the Android docs say that on API only `armeabi-v7a` is supported and others removed on NDK r17.

Rechi commented on tools/depends/configure.ac:
```diff
@@ -296,6 +312,11 @@ case $host in
+    if [[ $host =~ "i686"*"-linux-gnu"* ]] || [[ $host =~ "i"*"86"*"-"*"-linux-uclibc"* ]]; then
+      use_cmake_cpu=i686
+    else
+      use_cmake_cpu=x86_64
```
There is a variable which already contains exactly one of the two values, depending on the host used to configure. So why not use that one?

AlwinEsch replied:
I look about a regex (sed) to extract it.

AlwinEsch replied:
Thought you mean the `$host` and not come direct to them that `$host_cpu` is created from them.

But why we have so crazy switch compares about `$host` :smile:? 
A `*i686*-linux-gnu*` or `x86_64*-linux-gnu` say me that something can before or behind host cpu and not only a i686 possible :thinking:

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -286,6 +292,15 @@ case $host in
     fi
+    if [[[ "$host" =~ "arm64-" ]]] || [[[ "$host" =~ "aarch64-" ]]]; then
+      use_cmake_cpu=aarch64
+    elif [[[ "$host" =~ "arm32-" ]]] || [[[ "$host" =~ "aarch32-" ]]]; then
```
On the compare places was add of 3 "[" needed, the bootstrap strip one away and on this compare it need 2 "[[".

Rechi replied:
What is aarch32 and arm32?
There exist one variable which always contains the right value.

AlwinEsch replied:
Have added for this:
```
      # aarch32 related to https://en.wikipedia.org/wiki/ARM_architecture#AArch32 is v7-a
      # arm32 should never be there, only in the check to avoid possible wrong cmake CPU
```
**You mean to use the `$host_cpu` for compare?**
For them I'm sorry, that `$host_cpu` is created from `configure` itself from `$host` by `sed` like below I'm not seen direct.

It check above the host for `aarch64*-*-linux-gnu*|aarch*-*-linux-be*` with this `aarch*` other as 64 then also in `$host_cpu` not correct for cmake.

The `use_cmake_cpu=$(echo $host | sed -e 's/\-.*$//')` becomes changed to `use_cmake_cpu=$host_cpu`

AlwinEsch replied:
Have also looked now about this `arm*-*-linux-gnu*|arm*-*-linux-uclibc*|aarch64*-*-linux-gnu*|aarch*-*-linux-be*` in history.
The `aarch*-*-linux-be*` is thought for `aarch64` only, a bit confusing if also a aarch32 is used on the world.

See https://github.com/xbmc/xbmc/pull/10659/files#diff-25bb87ae2148a0a2abba0a8a4296b5bfR189

AlwinEsch reviewed (commented):
Is updated

AlwinEsch commented on tools/depends/configure.ac:
```diff
@@ -301,6 +316,11 @@ case $host in
     platform_os="linux"
     ffmpeg_options_default="--enable-vaapi --enable-vdpau --cpu=$use_cpu"
+    if [[[ "$host" =~ "i"."86" ]]]; then
+      use_cmake_cpu=${host:0:4}
```
Have changed to extract the string from host, then can also e.g. i386 be given.

Rechi replied:
There is no regex or other string manipulation needed. There exist one variable which always contains the right value.

AlwinEsch commented:
jenkins build this with addons please

Rechi commented on tools/depends/configure.ac:
```diff
@@ -437,11 +471,21 @@ case $use_platform in
+     platform_cflags="-fPIC -mcpu=cortex-a72 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad"
+     platform_cxxflags="-fPIC -mcpu=cortex-a72 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad"
+     platform_ldflags="-lpthread"
+     ;;
```
How is adding a new Linux platform related to this PR?
Have you runtime tested it?

AlwinEsch replied:
Was wrong added, only thought for test at home. Becomes removed.

AlwinEsch committed:
[build][depends] add support for x86_64-linux-android

Previously, only an "i686-linux-android" was used which actually only corresponds to
the 32bit CPU.

In reference to this https://developer.android.com/ndk/guides/other_build_systems at 64bit "x86_64-linux-android" is used.

AlwinEsch commented:
Have it cleaned, need to do tommorow a big check of them.

About this `case` values:
- `aarch64*-*linux-android*`
- `aarch64*-*-linux-gnu*`
- `aarch*-*-linux-be*`
- `*i686*-linux-gnu*`
- `i*86*-*-linux-uclibc*`
- `x86_64*-linux-gnu*`

Why they have a `*` before or behind `host_cpu` string? That is much confusing and shows mostly that something else can on the **CPU**.

AlwinEsch commented:
Tested on following OS and the `CMAKE_SYSTEM_PROCESSOR` was available in `./*/CMakeFiles/*/CMakeSystem.cmake` on addons and cmake depends (e.g. p8-platform, libandroidjni, libfmt, rapidjson, mariadb, flatbuffers, taglib).

Acceptable to bring?

**Own system:**
```
#------- configuration -------#
ccache:		 no
build type:	 release
toolchain:	 /usr
cpu:		 x86_64
cpu (cmake):	 x86_64
host:		 x86_64-linux-gnu
cflags:		 -fPIC -DPIC -DNDEBUG=1 -O2 
cxxflags:	 -fPIC -DPIC -DNDEBUG=1 -O2 
ldflags:	   
ffmpeg options:	 --enable-vaapi --enable-vdpau --cpu=x86_64
prefix:		 /home/alwin/own
depends:	 /home/alwin/own/x86_64-linux-gnu-release
```

**Android arm**:
```
#------- configuration -------#
ccache:		 no
build type:	 debug
toolchain:	 /home/alwin/Development/android-tools/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64
cpu:		 armeabi-v7a
cpu (cmake):	 armv7-a
host:		 arm-linux-android
cflags:		 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -march=armv7-a -mtune=cortex-a9 -mfloat-abi=softfp -mfpu=neon -Og -g -D_DEBUG 
cxxflags:	 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -march=armv7-a -mtune=cortex-a9 -mfloat-abi=softfp -mfpu=neon -frtti -Og -g -D_DEBUG 
ldflags:	 -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libatomic.a -L/home/alwin/Development/android-tools/xbmc-depends/arm-linux-android-21-debug/lib/android-21 -Wl,--exclude-libs,libunwind.a  
ffmpeg options:	 
prefix:		 /home/alwin/Development/android-tools/xbmc-depends
depends:	 /home/alwin/Development/android-tools/xbmc-depends/arm-linux-android-21-debug
ndk-api-level:	 21
build-tools:	 /home/alwin/Development/android-tools/android-sdk-linux/tools:/home/alwin/Development/android-tools/android-sdk-linux/platform-tools:/home/alwin/Development/android-tools/android-sdk-linux/build-tools/28.0.3
```

**Android arm64**:
```
#------- configuration -------#
ccache:		 no
build type:	 debug
toolchain:	 /home/alwin/Development/android-tools/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64
cpu:		 arm64-v8a
cpu (cmake):	 aarch64
host:		 aarch64-linux-android
cflags:		 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -march=armv8-a -mtune=cortex-a53 -Og -g -D_DEBUG 
cxxflags:	 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -march=armv8-a -mtune=cortex-a53 -frtti -Og -g -D_DEBUG 
ldflags:	 -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libatomic.a -L/home/alwin/Development/android-tools/xbmc-depends/aarch64-linux-android-21-debug/lib/android-21  
ffmpeg options:	 
prefix:		 /home/alwin/Development/android-tools/xbmc-depends
depends:	 /home/alwin/Development/android-tools/xbmc-depends/aarch64-linux-android-21-debug
ndk-api-level:	 21
build-tools:	 /home/alwin/Development/android-tools/android-sdk-linux/tools:/home/alwin/Development/android-tools/android-sdk-linux/platform-tools:/home/alwin/Development/android-tools/android-sdk-linux/build-tools/28.0.3
```

**Android x86**
```
#------- configuration -------#
ccache:		 no
build type:	 debug
toolchain:	 /home/alwin/Development/android-tools/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64
cpu:		 i686
cpu (cmake):	 i686
host:		 i686-linux-android
cflags:		 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -Og -g -D_DEBUG 
cxxflags:	 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -frtti -Og -g -D_DEBUG 
ldflags:	 -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libatomic.a -L/home/alwin/Development/android-tools/xbmc-depends/i686-linux-android-21-debug/lib/android-21  
ffmpeg options:	 
prefix:		 /home/alwin/Development/android-tools/xbmc-depends
depends:	 /home/alwin/Development/android-tools/xbmc-depends/i686-linux-android-21-debug
ndk-api-level:	 21
build-tools:	 /home/alwin/Development/android-tools/android-sdk-linux/tools:/home/alwin/Development/android-tools/android-sdk-linux/platform-tools:/home/alwin/Development/android-tools/android-sdk-linux/build-tools/28.0.3
```

**Android x86_64**
```
#------- configuration -------#
ccache:		 no
build type:	 debug
toolchain:	 /home/alwin/Development/android-tools/android-ndk-r21/toolchains/llvm/prebuilt/linux-x86_64
cpu:		 x86_64
cpu (cmake):	 x86_64
host:		 x86_64-linux-android
cflags:		 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -Og -g -D_DEBUG 
cxxflags:	 -DANDROID -D__ANDROID_API__=21 -fexceptions -funwind-tables -fstack-protector-strong -no-canonical-prefixes -fPIC -DPIC -frtti -Og -g -D_DEBUG 
ldflags:	 -Wl,--exclude-libs,libgcc.a -Wl,--exclude-libs,libatomic.a -L/home/alwin/Development/android-tools/xbmc-depends/x86_64-linux-android-21-debug/lib/android-21  
ffmpeg options:	 
prefix:		 /home/alwin/Development/android-tools/xbmc-depends
depends:	 /home/alwin/Development/android-tools/xbmc-depends/x86_64-linux-android-21-debug
ndk-api-level:	 21
build-tools:	 /home/alwin/Development/android-tools/android-sdk-linux/tools:/home/alwin/Development/android-tools/android-sdk-linux/platform-tools:/home/alwin/Development/android-tools/android-sdk-linux/build-tools/28.0.3
```

**Raspberry PI2**
```
#------- configuration -------#
ccache:		 no
build type:	 release
toolchain:	 /home/alwin/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf
cpu:		 cortex-a7
cpu (cmake):	 armv7-a
host:		 arm-linux-gnueabihf
cflags:		 -fPIC -mcpu=cortex-a7 -mfloat-abi=hard -mfpu=neon-vfpv4 -mvectorize-with-neon-quad -pipe -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -Wno-deprecated-declarations -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vcos/pthreads -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vmcs_host/linux -DNDEBUG=1 -Os 
cxxflags:	 -fPIC -mcpu=cortex-a7 -mfloat-abi=hard -mfpu=neon-vfpv4 -mvectorize-with-neon-quad -pipe -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -Wno-deprecated-declarations -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vcos/pthreads -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vmcs_host/linux -DNDEBUG=1 -Os 
ldflags:	 -lpthread -L/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/lib -lEGL -lGLESv2 -lbcm_host -lvcos    -lvchiq_arm  
ffmpeg options:	 --cpu=cortex-a7
prefix:		 /home/alwin/kodi-rpi
depends:	 /home/alwin/kodi-rpi/raspberry-pi2-release
```

**Raspberry PI3**
```
#------- configuration -------#
ccache:		 no
build type:	 release
toolchain:	 /home/alwin/Development/Kodi/_Raspberry/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf
cpu:		 cortex-a53
cpu (cmake):	 armv7-a
host:		 arm-linux-gnueabihf
cflags:		 -fPIC -mcpu=cortex-a53 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad -pipe -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -Wno-deprecated-declarations -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vcos/pthreads -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vmcs_host/linux -DNDEBUG=1 -Os 
cxxflags:	 -std=c++14 -fPIC -mcpu=cortex-a53 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad -pipe -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -Wno-deprecated-declarations -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vcos/pthreads -isystem/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/include/interface/vmcs_host/linux -DNDEBUG=1 -Os 
ldflags:	 -lpthread -L/home/alwin/Development/Kodi/_Raspberry/firmware/opt/vc/lib -lEGL -lGLESv2 -lbcm_host -lvcos    -lvchiq_arm  
ffmpeg options:	 --cpu=cortex-a53
prefix:		 /home/alwin/kodi-rpi4
depends:	 /home/alwin/kodi-rpi4/raspberry-pi3-release
```

**OSX:**
```
#------- configuration -------#
-e ccache:		 no
-e build type:	 debug
-e toolchain:	 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
-e cpu:		 x86_64
-e cpu (cmake):	 x86_64
-e host:		 x86_64-apple-darwin
-e cflags:		 -fheinous-gnu-extensions -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.13 -g -D_DEBUG 
-e cxxflags:	 -std=c++14 -no-cpp-precomp -arch x86_64 -mmacosx-version-min=10.13 -stdlib=libc++ -g -D_DEBUG 
-e ldflags:	 -Wl,-search_paths_first -arch x86_64 -mmacosx-version-min=10.13 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -stdlib=libc++  -liconv 
-e ffmpeg options:	 
-e prefix:		 /Users/Shared/xbmc-depends
-e depends:	 /Users/Shared/xbmc-depends/macosx10.14_x86_64-target-debug
```

**IOS 64bit:**
```
#------- configuration -------#
-e ccache:		 no
-e build type:	 debug
-e toolchain:	 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
-e cpu:		 arm64
-e cpu (cmake):	 aarch64
-e host:		 arm-apple-darwin
-e cflags:		 -fheinous-gnu-extensions -no-cpp-precomp -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -gdwarf-2 -Wno-error=implicit-function-declaration -arch arm64 -miphoneos-version-min=9.0 -g -D_DEBUG 
-e cxxflags:	 -std=c++14 -no-cpp-precomp -arch arm64 -miphoneos-version-min=9.0 -stdlib=libc++ -g -D_DEBUG 
-e ldflags:	 -Wl,-search_paths_first -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk/usr/lib -arch arm64 -miphoneos-version-min=9.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk -stdlib=libc++  -liconv 
-e ffmpeg options:	 
-e prefix:		 /Users/Shared/xbmc-depends
-e depends:	 /Users/Shared/xbmc-depends/iphoneos12.1_arm64-target-debug
```

**TVOS:**
```
#------- configuration -------#
-e ccache:		 no
-e build type:	 debug
-e toolchain:	 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
-e cpu:		 arm64
-e cpu (cmake):	 aarch64
-e host:		 arm-apple-darwin
-e cflags:		 -fheinous-gnu-extensions -no-cpp-precomp -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -gdwarf-2 -Wno-error=implicit-function-declaration -arch arm64 -mappletvos-version-min=11.0 -fembed-bitcode -g -D_DEBUG 
-e cxxflags:	 -std=c++14 -no-cpp-precomp -arch arm64 -mappletvos-version-min=11.0 -stdlib=libc++ -fembed-bitcode -g -D_DEBUG 
-e ldflags:	 -Wl,-search_paths_first -L/usr/lib -arch arm64 -mappletvos-version-min=11.0 -isysroot  -stdlib=libc++  -liconv 
-e ffmpeg options:	 
-e prefix:		 /Users/Shared/xbmc-depends
-e depends:	 /Users/Shared/xbmc-depends/appletvos11.0_arm64-target-debug
```

AlwinEsch commented:
jenkins build this with addons please

Rechi commented:
As long as this contains an unrelated commit (adding a new target platform), no.

AlwinEsch commented:
This is cherry picked from https://github.com/xbmc/xbmc/pull/17377 to have for test inside and to prevent rebase problem. The other comes first, if OK, then this.

AlwinEsch committed:
[depends] set CMAKE_SYSTEM_PROCESSOR on Toolchain_*.cmake.in files

Some addons and his dependencies built with cmake need the CMAKE_SYSTEM_PROCESSOR to prepare for the desired OS.
Further was this also not given to Kodi's own depends build

Rechi commented:
IMO the whole `use_cmake_cpu` special handling isn't needed. See #17385 for a much simpler solution.

AlwinEsch commented:
Here https://github.com/xbmc/xbmc/commit/6656fdfc9c80161202c5c0a6709669d96c2ce7ba it was partly by this way, was not correct but on right place then.

But a small thing does then not match on your request. I write a comment there.

AlwinEsch commented:
@rechi what do you think, take your path, close this and change the Leia backport to yours?

Rechi commented:
I prefer using `host_cpu` as value for `CMAKE_SYSTEM_PROCESSOR`. There is no need for being that precise about the architecture.
